### ISSUE TITLE
default rustfmt formatting will cause suspicious_else_formatting clippy warning

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#10273]
### Summary

.

### Reproducer

I tried this code:

```rust
pub fn main() {
    if true {
        /* hello */
    } else /* whelp */ {
        ; // yay, code
    }
}
```
This code compiles without clippy warnings, but rustformat will move the `whelp` comment in its own line and thus separate the ` else` from the `{`.
````rust
pub fn main() {
    if true {
        /* hello */
    } else
    /* whelp */
    { // yay, code
    }
}
````
Which will cause clippy to warn:
````
warning: this is an `else {..}` but the formatting might hide it
 --> src/main.rs:4:6
  |
4 |       } else
  |  ______^
5 | |     /* whelp */
6 | |     { // yay, code
  | |____^
  |
  = note: to remove this lint, remove the `else` or remove the new line between `else` and `{..}`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#suspicious_else_formatting
  = note: `#[warn(clippy::suspicious_else_formatting)]` on by default

````

I'm not sure what the best situation is here but warning on code that was "created" by rustfmt feels questionable.

Should the comment between `else` and `{` prevent linting in this case? 

### Version

```text
rustc 1.69.0-nightly (dc1d9d50f 2023-01-31)
binary: rustc
commit-hash: dc1d9d50fba2f6a1ccab8748a0050cde38253f60
commit-date: 2023-01-31
host: x86_64-unknown-linux-gnu
release: 1.69.0-nightly
LLVM version: 15.0.7
```
